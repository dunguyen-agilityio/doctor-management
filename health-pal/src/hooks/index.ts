export * from './use-app-loading'
