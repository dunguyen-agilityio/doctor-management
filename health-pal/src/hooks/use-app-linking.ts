import { useEffect } from 'react'

import * as Linking from 'expo-linking'
import { router } from 'expo-router'

import { ROUTES } from '@/constants'

export const useAppLinking = () => {
  useEffect(() => {
    const handleInitialURL = async () => {
      const initialUrl = Linking.getLinkingURL()
      if (initialUrl) {
        handleDeepLink(initialUrl)
      }
    }

    const subscription = Linking.addEventListener('url', ({ url }) => {
      handleDeepLink(url)
    })

    handleInitialURL()

    return () => subscription.remove()
  }, [])

  const handleDeepLink = (url: string) => {
    const { path, hostname, queryParams } = Linking.parse(url)

    const isDoctorDetails = hostname === 'doctors' && path?.includes('details')
    const isBooking = hostname === 'booking'
    const param = path?.split('/')[1] ?? ''

    console.log(
      'first',
      queryParams,
      ['doctorId', 'doctorDocId'].every((key) => (queryParams ?? {})[key]),
    )

    switch (true) {
      case isDoctorDetails:
        router.push({ pathname: ROUTES.DOCTOR, params: { id: param } })
        break

      case isBooking && ['doctorId', 'doctorDocId'].some((key) => !(queryParams ?? {})[key]):
        router.replace(ROUTES.NOT_FOUND)
        break

      default:
        break
    }
  }
}
