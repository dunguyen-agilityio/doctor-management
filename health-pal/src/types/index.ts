export * from './auth'
export * from './booking'
export * from './doctor'
export * from './favorite'
export * from './image'
export * from './modal'
export * from './route'
export * from './specialty'
export * from './strapi'
