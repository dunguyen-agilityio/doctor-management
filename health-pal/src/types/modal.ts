export type ModalRef<T = null> = { open: (params?: T) => void; close: () => void }
