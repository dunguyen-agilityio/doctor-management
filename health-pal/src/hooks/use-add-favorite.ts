import { useMutation } from '@tanstack/react-query'

import { useToastController } from '@tamagui/toast'

import { useRequireAuth } from '@app/contexts'
import { queryClient } from '@app/react-query.config'
import { addFavorite } from '@app/services/favorite'
import { FAVORITE_TYPES } from '@app/types/favorite'

export const useAddFavorite = (type: FAVORITE_TYPES, itemName: string) => {
  const { jwt, user } = useRequireAuth().session
  const { id: userId } = user
  const toast = useToastController()

  const handleMutation = (itemId: number) => {
    return addFavorite({ itemId, type }, userId, jwt)
  }

  return useMutation({
    mutationFn: handleMutation,
    onSuccess: () => {
      queryClient.invalidateQueries({ queryKey: ['favorites', type, userId] })
      toast.show('Added to Favorites', {
        message: `${itemName} has been added to your favorites.`,
        duration: 3000,
        type: 'success',
      })
    },
    onError: () => {
      toast.show('Action Failed', {
        message: 'Failed to add favorite. Please try again.',
        type: 'error',
        duration: 3000,
      })
    },
  })
}
