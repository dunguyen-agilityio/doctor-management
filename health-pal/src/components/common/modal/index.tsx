import { useEffect } from 'react'

import { Button, Dialog, DialogContentProps, DialogProps, Unspaced } from 'tamagui'

import { WINDOW_SIZE } from '@app/constants'

import { useModal } from '@app/hooks/use-modal'

import { X } from '@icons'

import { ModalRef } from '@app/types/modal'

export interface ModalProps extends DialogContentProps, Omit<DialogProps, 'children'> {
  closeButtonShown?: boolean
  ref?: React.Ref<ModalRef>
}

const Modal = ({
  children,
  closeButtonShown,
  ref,
  open: defaultOpen,
  ...props
}: React.PropsWithChildren<ModalProps>) => {
  const [open, setOpen] = useModal(ref)

  useEffect(() => {
    if (defaultOpen) {
      setOpen(true)
    }
  }, [defaultOpen, setOpen])

  return (
    <Dialog modal open={open} onOpenChange={setOpen}>
      <Dialog.Portal>
        <Dialog.Overlay
          testID="overlay"
          backgroundColor="$shadow3"
          animateOnly={['transform', 'opacity']}
          animation={[
            'quicker',
            {
              opacity: {
                overshootClamping: true,
              },
            },
          ]}
          enterStyle={{ opacity: 0, scale: 0.95 }}
          exitStyle={{ opacity: 0 }}
        />

        <Dialog.Content
          role="dialog"
          testID="content"
          padding={0}
          width={WINDOW_SIZE.width - 48}
          minHeight={WINDOW_SIZE.height / 2}
          borderRadius="$2"
          overflow="hidden"
          shadowColor="#fff"
          shadowOffset={{ height: 0, width: 0 }}
          key="content"
          animateOnly={['transform', 'opacity']}
          animation={[
            'quicker',
            {
              opacity: {
                overshootClamping: true,
              },
            },
          ]}
          enterStyle={{ x: 0, y: -20, opacity: 0 }}
          exitStyle={{ x: 0, y: 10, opacity: 0, scale: 0.95 }}
          {...props}>
          {children}
          {closeButtonShown && (
            <Unspaced>
              <Dialog.Close asChild>
                <Button
                  position="absolute"
                  top="$md"
                  right="$md"
                  size="$5"
                  circular
                  icon={X}
                  aria-label="Close the dialog"
                  accessibilityHint="Closes the cancellation dialog without making changes"
                  role="button"
                />
              </Dialog.Close>
            </Unspaced>
          )}
        </Dialog.Content>
      </Dialog.Portal>
    </Dialog>
  )
}

export default Modal
