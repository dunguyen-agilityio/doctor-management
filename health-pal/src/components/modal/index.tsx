import { useImperativeHandle, useState } from 'react'

import { X } from '@tamagui/lucide-icons'
import { Button, Dialog, DialogContentProps, DialogProps, Unspaced } from 'tamagui'

import { WINDOW_SIZE } from '@app/constants'

import { ModalRef } from '@app/types/modal'

interface ModalProps extends DialogProps {
  closeButtonShown?: boolean
  contentProps?: DialogContentProps
  ref?: React.Ref<ModalRef>
}

const Modal = ({
  children,
  closeButtonShown,
  contentProps,
  ref,
  ...props
}: React.PropsWithChildren<ModalProps>) => {
  const [open, setOpen] = useState(false)

  useImperativeHandle(ref, () => ({
    open: () => setOpen(true),
    close: () => setOpen(false),
  }))

  return (
    <Dialog modal open={open} onOpenChange={setOpen} {...props}>
      <Dialog.Portal>
        <Dialog.Overlay
          key="overlay"
          backgroundColor="$shadow3"
          animateOnly={['transform', 'opacity']}
          animation={[
            'quicker',
            {
              opacity: {
                overshootClamping: true,
              },
            },
          ]}
          enterStyle={{ opacity: 0, scale: 0.95 }}
          exitStyle={{ opacity: 0 }}
        />

        <Dialog.Content
          padding={0}
          width={WINDOW_SIZE.width - 48}
          minHeight={WINDOW_SIZE.height / 2}
          borderRadius="$2"
          overflow="hidden"
          shadowColor="#fff"
          shadowOffset={{ height: 0, width: 0 }}
          key="content"
          animateOnly={['transform', 'opacity']}
          animation={[
            'quicker',
            {
              opacity: {
                overshootClamping: true,
              },
            },
          ]}
          enterStyle={{ x: 0, y: -20, opacity: 0 }}
          exitStyle={{ x: 0, y: 10, opacity: 0, scale: 0.95 }}
          {...contentProps}>
          {children}
          {closeButtonShown && (
            <Unspaced>
              <Dialog.Close asChild>
                <Button position="absolute" top="$md" right="$md" size="$5" circular icon={X} />
              </Dialog.Close>
            </Unspaced>
          )}
        </Dialog.Content>
      </Dialog.Portal>
    </Dialog>
  )
}

export default Modal
