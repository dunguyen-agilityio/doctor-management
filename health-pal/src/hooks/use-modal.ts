import { useImperativeHandle, useState } from 'react'

import { ModalRef } from '@/types/modal'

export const useModal = (ref?: React.Ref<ModalRef | null>) => {
  const [open, setOpen] = useState(false)

  useImperativeHandle(ref, () => ({
    open: () => setOpen(true),
    close: () => setOpen(false),
  }))

  return [open, setOpen] as const
}
