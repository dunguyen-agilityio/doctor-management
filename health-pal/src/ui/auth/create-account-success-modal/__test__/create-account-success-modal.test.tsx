import { render, screen } from '@utils-test'

import * as MockModal from '@/hooks/use-modal'

import { ModalRef } from '@/types/modal'

import CreateAccountSuccessModal from '..'

describe('CreateAccountSuccessModal', () => {
  const mockRef: React.RefObject<ModalRef> = {
    current: {
      open: jest.fn(),
      close: jest.fn(),
    },
  }
  let setOpenMock: jest.Mock

  beforeEach(() => {
    jest.clearAllMocks()
    setOpenMock = jest.fn()
    jest.spyOn(MockModal, 'useModal').mockImplementation((ref) => {
      if (ref && ref !== null && 'current' in ref) {
        ref.current = mockRef.current
      }

      return [true, setOpenMock]
    })
  })

  afterEach(() => {
    jest.restoreAllMocks()
  })

  it('has correct accessibility attributes', () => {
    render(<CreateAccountSuccessModal ref={mockRef} />)

    const modal = screen.getByLabelText('Account created successfully')
    const loadingIndicator = screen.getByLabelText('Redirecting, please wait')

    expect(modal).toHaveProp('aria-label', 'Account created successfully')
    expect(loadingIndicator).toHaveProp('aria-label', 'Redirecting, please wait')
  })

  it('matches snapshot when open', () => {
    const { toJSON } = render(<CreateAccountSuccessModal ref={mockRef} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
