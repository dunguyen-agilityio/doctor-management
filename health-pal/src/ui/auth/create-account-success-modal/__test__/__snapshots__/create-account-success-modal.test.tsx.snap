// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccountSuccessModal matches snapshot when open 1`] = `
[
  <View
    aria-label="Notifications (F8)"
    pointerEvents="box-none"
    role="region"
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "maxWidth": "100%",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100000,
      }
    }
    tabIndex={-1}
  >
    <View
      focusable={false}
      pointerEvents="box-none"
      style={
        {
          "flexDirection": "column",
          "maxWidth": "100%",
          "position": "absolute",
        }
      }
    />
  </View>,
  <View
    pointerEvents="auto"
    style={
      {
        "alignItems": "center",
        "bottom": 0,
        "flexDirection": "column",
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      collapsable={false}
      pointerEvents="auto"
      style={
        {
          "backgroundColor": "rgba(0,0,0,0.16)",
          "bottom": 0,
          "flexDirection": "column",
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": [
            {
              "scale": 0.95,
            },
          ],
          "zIndex": 99999,
        }
      }
      testID="overlay"
    />
    <View
      aria-describedby="description-:rl:"
      aria-label="Account created successfully"
      aria-labelledby="title-:rl:"
      collapsable={false}
      id="content-:rl:"
      role="dialog"
      style={
        {
          "backgroundColor": "hsla(0, 0%, 100%, 1)",
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
          "flexDirection": "column",
          "minHeight": 406,
          "opacity": 0,
          "overflow": "hidden",
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "position": "relative",
          "shadowColor": "rgb(255,255,255)",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.0392156862745098,
          "shadowRadius": 10,
          "transform": [
            {
              "translateY": -20,
            },
            {
              "translateX": 0,
            },
            {
              "scale": 1,
            },
          ],
          "width": 327,
          "zIndex": 100000,
        }
      }
      testID="content"
    >
      <View
        style={[]}
      >
        <View
          style={[]}
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="73"
            bbWidth="73"
            fill="none"
            focusable={false}
            height="73"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 73,
                  "width": 73,
                },
              ]
            }
            vbHeight={73}
            vbWidth={73}
            width="73"
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                name="vuesax/bulk/shield-tick"
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  name="shield-tick"
                >
                  <RNSVGPath
                    d="M33.3799 6.68L16.8799 12.86C13.7299 14.06 11.1499 17.78 11.1499 21.17V45.47C11.1499 47.9 12.7399 51.11 14.6899 52.55L31.1899 64.88C34.0999 67.07 38.8699 67.07 41.7799 64.88L58.2799 52.55C60.2299 51.08 61.8199 47.9 61.8199 45.47V21.17C61.8199 17.81 59.2399 14.06 56.0899 12.89L39.5899 6.71C37.9099 6.05 35.0899 6.05 33.3799 6.68Z"
                    fill={
                      {
                        "payload": 4294967295,
                        "type": 0,
                      }
                    }
                    name="Vector"
                    propList={
                      [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M32.4797 43.19C31.9097 43.19 31.3397 42.98 30.8897 42.53L26.0597 37.7C25.1897 36.83 25.1897 35.39 26.0597 34.52C26.9297 33.65 28.3697 33.65 29.2397 34.52L32.4797 37.76L43.7897 26.45C44.6597 25.58 46.0997 25.58 46.9697 26.45C47.8397 27.32 47.8397 28.76 46.9697 29.63L34.0697 42.53C33.6197 42.98 33.0497 43.19 32.4797 43.19Z"
                    fill={
                      {
                        "payload": 4280888626,
                        "type": 0,
                      }
                    }
                    name="Vector_2"
                    propList={
                      [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          style={
            {
              "color": "#1c2a3a",
              "fontFamily": "Inter_600SemiBold",
              "fontSize": 20,
              "lineHeight": 30,
            }
          }
          suppressHighlighting={true}
        >
          Congratulations!
        </Text>
        <Text
          style={
            {
              "color": "#1c2a3a",
              "fontFamily": "Inter_400Regular",
              "fontSize": 14,
              "lineHeight": 21,
              "textAlign": "center",
            }
          }
          suppressHighlighting={true}
        >
          Your account is ready to use. You will be redirected to the Home Page in a few secondsâ€¦
        </Text>
        <View
          aria-label="Redirecting, please wait"
          role="status"
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "marginTop": 20,
              "width": "100%",
            }
          }
          testID="loading-indicator"
        >
          <View
            style={[]}
          >
            <View
              collapsable={false}
              style={
                {
                  "transform": [
                    {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="57"
                bbWidth="57"
                fill="none"
                focusable={false}
                height="57"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": 57,
                      "width": 57,
                    },
                  ]
                }
                testID="loader-icon"
                vbHeight={57}
                vbWidth={57}
                width="57"
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    [
                      "fill",
                    ]
                  }
                >
                  <RNSVGGroup
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  >
                    <RNSVGPath
                      d="M33.0506 5.75226C33.6442 3.23736 32.0867 0.717454 29.5718 0.123884C27.0569 -0.469685 24.537 1.08786 23.9434 3.60275C23.3498 6.11765 24.9074 8.63755 27.4223 9.23112C29.9372 9.82469 32.4571 8.26715 33.0506 5.75226Z"
                      fill={
                        {
                          "payload": 4280232503,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "fill",
                        ]
                      }
                    />
                    <RNSVGPath
                      d="M33.121 53.074C33.535 50.5234 31.803 48.1201 29.2524 47.7061C26.7017 47.2921 24.2984 49.0241 23.8844 51.5747C23.4704 54.1253 25.2024 56.5286 27.753 56.9426C30.3037 57.3567 32.707 55.6246 33.121 53.074Z"
                      fill={
                        {
                          "payload": 4280232503,
                          "type": 0,
                        }
                      }
                      opacity={0.4}
                      propList={
                        [
                          "fill",
                        ]
                      }
                    />
                    <RNSVGPath
                      d="M52.7054 33.1644C55.2806 32.9513 57.1955 30.691 56.9824 28.1158C56.7693 25.5406 54.509 23.6257 51.9338 23.8388C49.3586 24.0519 47.4437 26.3122 47.6568 28.8874C47.8699 31.4626 50.1302 33.3775 52.7054 33.1644Z"
                      fill={
                        {
                          "payload": 4280232503,
                          "type": 0,
                        }
                      }
                      opacity={0.1}
                      propList={
                        [
                          "fill",
                        ]
                      }
                    />
                    <RNSVGPath
                      d="M9.22941 29.5836C9.82869 27.0701 8.27687 24.5466 5.76332 23.9473C3.24978 23.3481 0.726347 24.8999 0.127073 27.4134C-0.472201 29.927 1.07962 32.4504 3.59316 33.0497C6.1067 33.649 8.63014 32.0971 9.22941 29.5836Z"
                      fill={
                        {
                          "payload": 4280232503,
                          "type": 0,
                        }
                      }
                      opacity={0.7}
                      propList={
                        [
                          "fill",
                        ]
                      }
                    />
                    <RNSVGPath
                      d="M52.2933 43.9649C54.1204 42.1377 54.1204 39.1753 52.2933 37.3481C50.4661 35.521 47.5037 35.521 45.6765 37.3481C43.8494 39.1753 43.8494 42.1377 45.6765 43.9649C47.5037 45.792 50.4661 45.792 52.2933 43.9649Z"
                      fill={
                        {
                          "payload": 4280232503,
                          "type": 0,
                        }
                      }
                      opacity={0.2}
                      propList={
                        [
                          "fill",
                        ]
                      }
                    />
                    <RNSVGPath
                      d="M8.71832 20.9674C11.2734 20.5792 13.0299 18.1932 12.6417 15.6381C12.2534 13.0831 9.86743 11.3265 7.31238 11.7148C4.75732 12.103 3.00077 14.489 3.38901 17.0441C3.77725 19.5991 6.16326 21.3557 8.71832 20.9674Z"
                      fill={
                        {
                          "payload": 4280232503,
                          "type": 0,
                        }
                      }
                      opacity={0.8}
                      propList={
                        [
                          "fill",
                        ]
                      }
                    />
                    <RNSVGPath
                      d="M11.3226 43.9649C13.1497 42.1377 13.1497 39.1753 11.3226 37.3481C9.49539 35.521 6.53298 35.521 4.70582 37.3481C2.87867 39.1753 2.87867 42.1377 4.70582 43.9649C6.53298 45.792 9.4954 45.792 11.3226 43.9649Z"
                      fill={
                        {
                          "payload": 4280232503,
                          "type": 0,
                        }
                      }
                      opacity={0.6}
                      propList={
                        [
                          "fill",
                        ]
                      }
                    />
                    <RNSVGPath
                      d="M51.3715 20.3671C52.1674 19.8946 52.8053 19.1967 53.2045 18.3617C53.6037 17.5266 53.7463 16.592 53.6143 15.6759C53.4822 14.7598 53.0815 13.9034 52.4627 13.2151C51.8439 12.5267 51.035 12.0373 50.138 11.8088C49.2411 11.5803 48.2966 11.6229 47.4239 11.9312C46.5512 12.2395 45.7895 12.7997 45.2352 13.5409C44.6809 14.2822 44.3588 15.1711 44.3098 16.0954C44.2608 17.0197 44.4869 17.9377 44.9596 18.7335C45.5933 19.8004 46.6248 20.5719 47.8272 20.8782C49.0297 21.1846 50.3046 21.0008 51.3715 20.3671Z"
                      fill={
                        {
                          "payload": 4280232503,
                          "type": 0,
                        }
                      }
                      opacity={0.05}
                      propList={
                        [
                          "fill",
                        ]
                      }
                    />
                    <RNSVGPath
                      d="M20.9295 51.5655C21.3824 50.7583 21.5856 49.835 21.5137 48.9122C21.4417 47.9895 21.0977 47.1088 20.5252 46.3816C19.9527 45.6544 19.1774 45.1133 18.2974 44.8267C17.4173 44.5401 16.4721 44.5209 15.5811 44.7716C14.6902 45.0222 13.8935 45.5314 13.292 46.2348C12.6905 46.9382 12.3111 47.8042 12.2017 48.7233C12.0924 49.6423 12.258 50.5732 12.6777 51.3981C13.0974 52.223 13.7523 52.9049 14.5596 53.3576C15.642 53.9644 16.9211 54.1165 18.1156 53.7804C19.3101 53.4444 20.3223 52.6477 20.9295 51.5655Z"
                      fill={
                        {
                          "payload": 4280232503,
                          "type": 0,
                        }
                      }
                      opacity={0.5}
                      propList={
                        [
                          "fill",
                        ]
                      }
                    />
                    <RNSVGPath
                      d="M42.1949 11.9344C44.5196 10.8061 45.4894 8.00699 44.3611 5.68234C43.2329 3.35769 40.4337 2.38785 38.109 3.51614C35.7844 4.64442 34.8146 7.44358 35.9428 9.76823C37.0711 12.0929 39.8703 13.0627 42.1949 11.9344Z"
                      fill={
                        {
                          "payload": 4280232503,
                          "type": 0,
                        }
                      }
                      opacity={0.01}
                      propList={
                        [
                          "fill",
                        ]
                      }
                    />
                    <RNSVGPath
                      d="M36.0705 51.5655C35.6176 50.7583 35.4144 49.835 35.4863 48.9122C35.5583 47.9895 35.9023 47.1088 36.4748 46.3816C37.0473 45.6544 37.8226 45.1133 38.7026 44.8267C39.5827 44.5401 40.528 44.5209 41.4189 44.7716C42.3099 45.0222 43.1065 45.5314 43.708 46.2348C44.3095 46.9382 44.689 47.8042 44.7983 48.7233C44.9076 49.6423 44.742 50.5732 44.3223 51.3981C43.9026 52.223 43.2477 52.9049 42.4404 53.3576C41.358 53.9644 40.0789 54.1165 38.8844 53.7804C37.6899 53.4444 36.6777 52.6477 36.0705 51.5655Z"
                      fill={
                        {
                          "payload": 4280232503,
                          "type": 0,
                        }
                      }
                      opacity={0.3}
                      propList={
                        [
                          "fill",
                        ]
                      }
                    />
                    <RNSVGPath
                      d="M17.8818 12.2916C20.4018 11.72 21.9812 9.21378 21.4096 6.69381C20.838 4.17383 18.3318 2.59435 15.8118 3.16595C13.2919 3.73755 11.7124 6.24377 12.284 8.76375C12.8556 11.2837 15.3618 12.8632 17.8818 12.2916Z"
                      fill={
                        {
                          "payload": 4280232503,
                          "type": 0,
                        }
                      }
                      opacity={0.9}
                      propList={
                        [
                          "fill",
                        ]
                      }
                    />
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;
